### ISSUE TITLE
Spikegenerator period argument not working

### ISSUE THREAD

**Issue by denisalevi** [brian-team/brian2cuda#48]
These tests are failing:
```
brian2.tests.test_spikegenerator.test_spikenerator_period
brian2.tests.test_spikegenerator.test_spikenerator_rounding_period
brian2.tests.test_spikegenerator.test_spikegenerator_change_spikes
```
It looks like the generated spikes are not repeated after `period` time has passed.

**Comment by denisalevi** [comment]
The spikegenerator template uses `__syncthreads()` calls in a for loop, where not all threads execute the same number of threads. This can be a problem and might be associated with some of the failing tests here.
[This stackoverflow post](http://stackoverflow.com/a/5128818/6067651) shows how the loop should be implemented instead.

**Comment by denisalevi** [comment]
Started fixing this issue in PR #186 (priod argument works in simple cases now, but some tests still faile). PR to continue work on this is #194 .

**Comment by denisalevi** [comment]
This issue is fixed with PR #223 (which will be merged into master with PR #217 )

These two tests are still failing in the test suite, but this is due to a bug in the spikemonitor (#221 ) and not due to the spikegenerator (tests pass when using separate spikemonitors per run).
```
brian2.tests.test_spikegenerator.test_spikegenerator_change_spikes
brian2.tests.test_spikegenerator.test_spikegenerator_change_period
```
